### DISCUSSION TITLE
Hono RPC VS tRPC in terms of performance?

### DISCUSSION THREAD

**Discussion by Blankeos** [honojs/hono#2501] [General]
Although, I don't completely know the nuanced differences between the two... If it's just end-to-end typesafety, Hono's dev experience is the best for me. 

It doesn't mask the request in a jibberish URL like tRPC does, so it's actually also still a REST-usable API. I love it so much that I don't feel the need to use tRPC anymore. But is it really a viable option to just completely replace tRPC from my stack in favor of Hono?

---

Do you guys have any recommendations on how to benchmark the two? I'd love to try comparing the performance between the two. 

I'd love to measure the following, but don't know how:
- node module size
- performance (speed or number of requests handled)

Any ideas?

**Comment by NamesMT** [comment]
TL;DR: If you use trpc, in production app you would likely always piggy-back it on another framework, so the performance I guess would be ~ that framework, for the size, it would be that framework + trpc, so definitely won't be small.

---
For detailed benchmarking you could try cloning, modify & test these two repos:
https://github.com/SaltyAom/bun-http-framework-benchmark
https://github.com/fastify/benchmarks

Would also be interested in the result :D, I'm too lazy rn to run it.

IMHO Hono feels like the sweet spot of performance, functionality, DX right now.
100% thumbs up to use Hono in your stack.
